firstvoices/backend/serializers/site_data_serializers.py

```diff
@@ -0,0 +1,188 @@
+    match type:
+        case DictionaryEntry.TypeOfDictionaryEntry.WORD:
+            return "words"
+        case DictionaryEntry.TypeOfDictionaryEntry.PHRASE:
+            return "phrases"
+        case _:
+            return None
+
+
+class SiteDataSerializer(SiteContentLinkedTitleSerializer):
+    site_data_export = serializers.SerializerMethodField()
+
+    def get_site_data_export(self, site):
+        request = self.context.get("request")
+        characters_list = [
+            character
+            for character in utils.filter_by_viewable(
+                request.user, site.character_set.all()
+            ).order_by("sort_order")
+        ]
+        config = {
+            "L1": {
+                "name": None if site is None else site.title,
+                "lettersInLanguage": [character.title for character in characters_list],
+                "transducers": {},
+            },
+            "L2": {"name": "English"},
+            "build": datetime.now().strftime("%Y%m%d%H%M"),
+        }
+        queryset = (
+            site.dictionaryentry_set
+            if site is not None and site.dictionaryentry_set is not None
+            else None
+        )
+        request = self.context.get("request")
+        dictionary_entries = DictionaryEntryDataSerializer(
+            queryset, many=True, context={"request": request}
+        ).data
+
+        return {"config": config, "data": dictionary_entries}
+
+    class Meta:
+        model = Site
+        fields = ("site_data_export",)
+
+
+class CategoriesDataSerializer(serializers.ModelSerializer):
+    category = serializers.SerializerMethodField()
+    parent_category = serializers.SerializerMethodField()
+
+    def get_category(self, category):
+        return category.title
+
+    def get_parent_category(self, category):
+        if category.parent is not None:
+            return category.parent.title
+        else:
+            return None
+
+    class Meta:
+        model = Category
+        fields = (
+            "category",
+            "parent_category",
+        )
+
+
+class DictionaryEntryDataSerializer(serializers.ModelSerializer):
+    source = serializers.SerializerMethodField()
+    entryID = serializers.SerializerMethodField()
+    word = serializers.SerializerMethodField()
+    definition = serializers.SerializerMethodField()
+    audio = serializers.SerializerMethodField()
+    img = serializers.SerializerMethodField()
+    theme = CategoriesDataSerializer(source="categories", many=True)
+    secondary_theme = serializers.SerializerMethodField()
+    optional = serializers.SerializerMethodField()
+    compare_form = serializers.SerializerMethodField()
+    sort_form = serializers.SerializerMethodField()
+    sorting_form = serializers.SerializerMethodField()
+
+    def get_source(self, dictionaryentry):
+        return dict_entry_type_mtd_conversion(dictionaryentry.type)
+
+    def get_entryID(self, dictionaryentry):
+        return dictionaryentry.id
```

Comment:
These ones that are returning strings could alternately be defined as CharField(source='attribute_name')